src/MCPClient/lib/clientScripts/parse_dataverse_mets.py

```diff
@@ -0,0 +1,291 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""Parse the transfer METS file created as part of a Dataverse transfer and
+validate against the objects expected to be part of the SIP generated during
+transfer.
+"""
+
+import json
+import os
+import uuid
+
+# databaseFunctions requires Django to be set up
+import django
+django.setup()
+from django.utils import timezone
+
+# archivematicaCommon
+from archivematicaFunctions import get_file_checksum
+from custom_handlers import get_script_logger
+import databaseFunctions
+from main.models import Agent, File
+import metsrw
+from six.moves.urllib.parse import unquote_plus, urlparse, urlunparse
+
+logger = get_script_logger("archivematica.mcp.client.parse_dataverse_mets")
+transfer_objects_directory = "%transferDirectory%objects"
+
+
+class ParseDataverseError(Exception):
+    """Exception class for failures that might occur during the execution of
+    this script.
+    """
+
+
+def urldecode(url):
+    """Decode percent encoding introduced per RFC3986 when the Dataverse METS
+    was first written in this workflow:
+    https://tools.ietf.org/html/rfc3986#section-2.1.
+    """
```

Comment:
Did you mean to use that URL? I'm confused about the word you used: "workflow".

What would happen if urldecode raises `ValueError`? E.g.:

```python
>>> urlparse("http://foo[bar.com/hello")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.6/urllib/parse.py", line 368, in urlparse
    splitresult = urlsplit(url, scheme, allow_fragments)
  File "/usr/lib/python3.6/urllib/parse.py", line 418, in urlsplit
    raise ValueError("Invalid IPv6 URL")
ValueError: Invalid IPv6 URL
```